Sources/TuistLoader/Loaders/ConfigLoader.swift

```diff
@@ -83,11 +83,19 @@ public final class ConfigLoader: ConfigLoading {
     }
 
     private func defaultConfig(at path: AbsolutePath) async throws -> Tuist {
-        let anyXcodeProjectOrWorkspace = !(
+        let anyNonTuistGeneratedXcodeProjectOrWorkspace = !(
             try await fileSystem.glob(directory: path, include: ["*.xcodeproj", "*.xcworkspace"])
                 .collect()
+                .concurrentFilter { path in
+                    let workspaceManifestExists = try await self.fileSystem
+                        .exists(path.parentDirectory.appending(component: "Workspace.swift"))
+                    let projectManifestExists = try await self.fileSystem
+                        .exists(path.parentDirectory.appending(component: "Project.swift"))
+                    return !workspaceManifestExists && !projectManifestExists
```

Comment:
Let me know if you'd do this differently.

Reply:
I think I'd restructure this a tiny bit to something like:
```swift
let hasXcodeProjectOrWorkspace =  !(try await fileSystem.glob(directory: path, include: ["*.xcodeproj", "*.xcworkspace"]).collect().isEmpty)
let hasWorkspaceOrProjectManifest = ...
if hasXcodeProjectOrWorkspace && !hasWorkspaceOrProjectManifest {
 return Tuist(project :.xcode())
} else {
  return TuistCore.Tuist.default
}
```